src/vs/workbench/contrib/tasks/browser/taskTerminalStatus.ts

```diff
@@ -8,32 +8,34 @@ import { Codicon } from 'vs/base/common/codicons';
 import { Disposable } from 'vs/base/common/lifecycle';
 import Severity from 'vs/base/common/severity';
 import { AbstractProblemCollector, StartStopProblemCollector } from 'vs/workbench/contrib/tasks/common/problemCollectors';
-import { TaskEvent, TaskEventKind } from 'vs/workbench/contrib/tasks/common/tasks';
+import { TaskEvent, TaskEventKind, TaskRunType } from 'vs/workbench/contrib/tasks/common/tasks';
 import { ITaskService, Task } from 'vs/workbench/contrib/tasks/common/taskService';
 import { ITerminalInstance } from 'vs/workbench/contrib/terminal/browser/terminal';
 import { ITerminalStatus } from 'vs/workbench/contrib/terminal/browser/terminalStatusList';
 import { MarkerSeverity } from 'vs/platform/markers/common/markers';
-import { spinningLoading } from 'vs/platform/theme/common/iconRegistry';
 
 interface TerminalData {
 	terminal: ITerminalInstance;
+	task: Task;
 	status: ITerminalStatus;
 	problemMatcher: AbstractProblemCollector;
+	taskRunEnded?: boolean;
```

Comment:
Should this be `boolean` rather than `boolean | undefined` to reduce the amount of possible states?